apps/nextra/pages/en/build/move/move-on-aptos/scripts/script-tutorial.mdx

```diff
@@ -0,0 +1,153 @@
+
+For more information about scripts, see the [Move scripts docs](./index.md)
+
+## Example use case
+
+The following example calls functions on the [aptos_coin.move](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/aptos_coin.move) module to confirm the balance of the destination account is less than `desired_balance`, and if so, tops it up to `desired_balance`.
+
+```move
+script {
+    use std::signer;
+    use aptos_framework::aptos_account;
+    use aptos_framework::aptos_coin;
+    use aptos_framework::coin;
+
+    fun main(src: &signer, dest: address, desired_balance: u64) {
+        let src_addr = signer::address_of(src);
+
+        addr::my_module::do_nothing();
+
+        let balance = coin::balance<aptos_coin::AptosCoin>(src_addr);
+        if (balance < desired_balance) {
+            aptos_account::transfer(src, dest, desired_balance - balance);
+        };
+    }
+}
+```
+
+## Execution
+
+Now that you know what you would like to accomplish, you need to determine:
+
+- Where do I put these files?
+- What do I name them?
+- Do I need a `Move.toml`?
+- How do I run my script with the CLI?
+
+Let us run through how to execute a Move script with a step-by-step example using the [Aptos CLI](../../../tools/aptos-cli/use-cli/index.md).
+
+1. Make a new directory for your work:
+
+   ```sh
+   mkdir testing
+   cd testing
+   ```
+
+2. Set up the Aptos CLI and [create an account](../../../tools/aptos-cli/use-cli/index.md):
+
+   ```sh
+   aptos init --network devnet
+   ```
+
+   You may reuse an existing private key (which looks like this: `0xbd944102bf5b5dfafa7fe865d8fa719da6a1f0eafa3cd600f93385482d2c37a4`), or it can generate a new one for you, as part of setting up your account. Let's say your account looks like the example below:
+
+   ```sh
+   ---
+   profiles:
+     default:
+       private_key: "0xbd944102bf5b5dfafa7fe865d8fa719da6a1f0eafa3cd600f93385482d2c37a4"
+       public_key: "0x47673ec83bb254cc9a8bfdb31846daacd0c96fe41f81855462f5fc5306312b1b"
+       account: cb265645385819f3dbe71aac266e319e7f77aed252cacf2930b68102828bf615
+       rest_url: "https://api.devnet.aptoslabs.com"
+       faucet_url: "https://faucet.devnet.aptoslabs.com"
+   ```
+
+3. From this same directory, initialize a new Move project:
+
+   ```sh
+   aptos move init --name run_script
+   ```
+
+4. Create a `my_script.move` file containing the example script above in a `sources/` subdirectory of your `testing/` directory. Also, create a `my_module.move` file as seen in the example below:
+
+   ```move
+   module addr::my_module {
+       public entry fun do_nothing() { }
+   }
+   ```
+
+   This results in the following file structure:
+
+   ```text
+   testing/
+      Move.toml
+      sources/
+         my_script.move
+         my_module.move
+   ```
```

Comment:
Can we use `<Filetree />` here (https://preview.aptos.dev/en/docs/components/filetree#usage)